src/manifest.rs

```diff
@@ -45,7 +45,39 @@ fn read_cargo_toml(path: &str) -> Result<CargoManifest, Error> {
     let mut cargo_contents = String::new();
     cargo_file.read_to_string(&mut cargo_contents)?;
 
-    Ok(toml::from_str(&cargo_contents)?)
+    let contents: toml::Value = toml::from_str(&cargo_contents)?;
+    let contents = &contents["package"];
+
+    let maybe_missing_fields = vec![
+        contents.get("description").is_none(),
+        contents.get("license").is_none(),
+        contents.get("repository").is_none(),
+    ];
+
+    if maybe_missing_fields.contains(&true) {
+        let mut error_buffer = String::new();
+        if maybe_missing_fields[0] {
+            error_buffer.push_str(&format!(
+                "  [ERROR] {} Missing field `description` for key `package`\n",
```

Comment:
these error messages are highly redundant, perhaps we have a function to make the message so we don't have to repeat it?

Reply:
we also should add that this field is missing in the `Cargo.toml` as it's not entirely clear that that is what this means.

Reply:
Yeah let's make it clear then!